slsim/lens_pop.py

```diff
@@ -137,6 +140,24 @@ def __init__(
                 sky_area=sky_area,
             )
 
+        elif deflector_type == "halo-models":
+            from slsim.Deflectors.compound_lens_halos_galaxies import (
+                CompoundLensHalosGalaxies,
+            )
+            from slsim.Pipelines.sl_hammocks_pipeline import SLHammocksPipeline
+
+            halo_galaxy_list = SLHammocksPipeline(
+                slhammocks_config=slhammocks_config, sky_area=sky_area, cosmo=cosmo
```

Comment:
If someone choose deflector_type == "halo-models" and slhammocks_config = None, then I think you should put slhammocks_config = "slsim/data/SL-Hammocks/pop_salp_a0001_zl001_to_5_wo_sub.csv" else: slhammocks_config = slhammocks_config. Meaning we should provide a default path if data is already strored inside the code.

Reply:
Now, if slhammocks_config == None, we are trying to create halo and galaxy populations not using the csv file. Does this mean that we should have a separate switch for generating halo and galaxy populations, separately from slhammocks_config?

Reply:
@kta-cosmo , thanks for the clarification. My comment was not relevant here. What you have done is correct. No need to change anything.